gateway/webhook/webhook.go

```diff
@@ -289,53 +295,53 @@ func (bt *batchWebhookTransformerT) batchTransformLoop() {
 		// stats
 		bt.stats.sourceStats[breq.sourceType].sourceTransform.End()
 
+		if len(batchResponse.responses) != len(payloadArr) {
+			batchResponse.batchError = errors.New("webhook batchtransform response events size does not equal sent events size")
+			pkgLogger.Errorf("%w", batchResponse.batchError)
+		}
 		if batchResponse.batchError != nil {
 			statusCode := http.StatusInternalServerError
 			if batchResponse.statusCode != 0 {
 				statusCode = batchResponse.statusCode
 			}
 			for _, req := range breq.batchRequest {
-				req.done <- webhookErrorRespT{statusCode: statusCode, err: batchResponse.batchError.Error()}
+				req.done <- transformerResponseT{StatusCode: statusCode, Err: batchResponse.batchError.Error()}
 			}
 			continue
 		}
 
-		if len(batchResponse.responses) != len(payloadArr) {
-			panic("webhook batchtransform response events size does not equal sent events size")
-		}
-
 		bt.stats.sourceStats[breq.sourceType].numOutputEvents.Count(len(batchResponse.responses))
 
 		for idx, resp := range batchResponse.responses {
 			webRequest := webRequests[idx]
-			output := resp.output
-			if resp.err != "" {
-				webRequests[idx].done <- webhookErrorRespT{err: resp.err, statusCode: resp.statusCode}
-				continue
+			if resp.Err == "" && resp.Output != nil {
```

Comment:
Can we introduce a set of integration tests for the webhook RequestHandler, using an httptest Server for mocking the transformer?
It should be possible to test and verify all three major scenarios (error, success, response to client)

The only way to catch up, is by start writing these missing test scenarios when we are touching some uncovered functionality.